powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationHistoryServiceBehavior.java

```diff
@@ -68,14 +69,27 @@ public void saveActivationAndLogChange(ActivationRecordEntity activation) {
      * @param activation Activation.
      * @param externalUserId User ID of user who caused the change.
      */
-    public void saveActivationAndLogChange(ActivationRecordEntity activation, String externalUserId) {
+    public void saveActivationAndLogChange(ActivationRecordEntity activation,  String externalUserId) {
+        saveActivationAndLogChange(activation, externalUserId, null);
+    }
+
+    /**
+     * Log activation status change into activation history.
+     *
+     * @param activation Activation.
+     * @param externalUserId User ID of user who caused the change.
+     * @param historyEventReason Optional reason, why this activation save event happened.
+     */
+    public void saveActivationAndLogChange(ActivationRecordEntity activation, String externalUserId, String historyEventReason) {
         Date changeTimestamp = new Date();
         activation.setTimestampLastChange(changeTimestamp);
         ActivationHistoryEntity activationHistoryEntity = new ActivationHistoryEntity();
         activationHistoryEntity.setActivation(activation);
         activationHistoryEntity.setActivationStatus(activation.getActivationStatus());
         if (activation.getActivationStatus() == ActivationStatus.BLOCKED) {
             activationHistoryEntity.setBlockedReason(activation.getBlockedReason());
+        } else {
+            activationHistoryEntity.setBlockedReason(historyEventReason);
```

Comment:
Let's consider renaming this column to something more fitting (i.e., `change_reason`), or maybe just open an issue now so that we do this change in some upcoming release.

Reply:
Let's rename that column in separate ticket. I've fixed that missing axis methods.